### PR TITLE
Config travis ci & disabled stylelint rules

### PR DESCRIPTION
**Pull request by Matt-Butler**
### Summary
Add travis-ci configuration file.

Running ruby and node tests.

Removed some Stylelint rules. (No naming convention in the application)

### CODE REVIEW

File: .stylelintrc [hunk 1]
```diff
@@ -2,12 +2,7 @@
   "extends": "stylelint-config-sass-guidelines",
   "rules": {
     "max-nesting-depth": 2,
-    "scss/at-mixin-pattern": "^(terra-)[a-z]+([a-z0-9-]+[a-z0-9]+)?$",
```
Comment (bjankord) [inline .stylelintrc]: I'd prefer to keep these rules as is and instead disable them inline where the custom site styles are.

If you add the following before the `.tui-html-code` selector in `terra-ui.com/app/modules/terra-ui.scss`, it should lint without erroring out on the custom site styles selector pattern.

```
// Custom site styles not follow terra selector-class-pattern
// stylelint-disable selector-class-pattern
// background/padding on code
.tui-html-code {
  background-color: #f7f7f9;
  padding: 1rem;
}
```

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+language: ruby
+
+env:
+  - TRAVIS_NODE_VERSION="6.3.0"
```
Comment (gneatgeek) [inline .travis.yml]: If we set this to 6 will it take the latest version of 6.X?

**Comment by gneatgeek** [comment]
👍

File: .travis.yml (hunk 2 above)
Comment (Matt-Butler) [inline .travis.yml]: Sorry, I forgot to comment on this earlier. This configuration is using nvm and not travis-ci convention found in the component repos. Therefore we need to target a specific version of node.

**Pull request merged**